cmd/cli/certificate_rotate.go

```diff
@@ -0,0 +1,352 @@
+		if err != nil {
+			return nil, err
+		}
+		obj, _, err := decode([]byte(file), nil, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		newMRC, ok := obj.(*v1alpha2.MeshRootCertificate)
+		if !ok {
+			return nil, fmt.Errorf("file provided is not recognized as MeshRootCertificate")
+		}
+		if newMRC.Spec.Intent != v1alpha2.InactiveIntent {
+			return nil, fmt.Errorf("file provided must be in Inactive role")
+		}
+
+		existing, err := r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Get(context.Background(), newMRC.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("unable to verify that MeshRootCertificate [%s] doesn't already exist. Error: %v", newMRC.Name, err)
+		}
+		if existing != nil {
+			return nil, fmt.Errorf("cannot use MeshRootCertificate [%s] as it already exists in cluster", newMRC.Name)
+		}
+
+		return r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Create(context.Background(), newMRC, metav1.CreateOptions{})
+	}
+
+	return r.createFromExisting(mrc)
+}
+
+func (r *rotateCmd) createFromExisting(mrc *v1alpha2.MeshRootCertificate) (*v1alpha2.MeshRootCertificate, error) {
+	switch {
+	case mrc.Spec.Provider.Tresor != nil:
+		return r.createTressorFromExisting(mrc)
+	case mrc.Spec.Provider.Vault != nil:
+		// since we can't create the secrets in vault we require them to create a file
+		return nil, fmt.Errorf("please use -file option to provide a MeshRootCertificate Vault provider")
+	case mrc.Spec.Provider.CertManager != nil:
+		// since we require them to set up cert-manager.io ahead of time we can create it for them
+		return nil, fmt.Errorf("please use -file option to provide a MeshRootCertificate for CertManager provider")
+	default:
+		return nil, fmt.Errorf("unknown certificate provider: %+v", mrc.Spec.Provider)
+	}
+}
+
+func generateName() string {
+	var chars = []rune("abcdefghijklmnopqrstuvwxyz1234567890")
+
+	s := make([]rune, 8)
+	for i := range s {
+		num, _ := rand.Int(rand.Reader, big.NewInt(int64(len(chars))))
+		s[i] = chars[num.Int64()]
+	}
+	return fmt.Sprintf("%s-%x", constants.DefaultMeshRootCertificateName, string(s))
+}
+
+func (r *rotateCmd) createTressorFromExisting(mrc *v1alpha2.MeshRootCertificate) (*v1alpha2.MeshRootCertificate, error) {
+	newName := generateName()
+	trustDomain := mrc.Spec.TrustDomain
+	if r.newTrustDomain != "" {
+		trustDomain = r.newTrustDomain
+	}
+	return r.configClient.ConfigV1alpha2().MeshRootCertificates(settings.Namespace()).Create(
+		context.Background(), &v1alpha2.MeshRootCertificate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      newName,
+				Namespace: settings.Namespace(),
+			},
+			Spec: v1alpha2.MeshRootCertificateSpec{
+				TrustDomain: trustDomain,
+				Intent:      v1alpha2.InactiveIntent,
+				Provider: v1alpha2.ProviderSpec{
+					Tresor: &v1alpha2.TresorProviderSpec{
+						CA: v1alpha2.TresorCASpec{
+							SecretRef: v1.SecretReference{
+								Name:      newName,
```

Comment:
Maybe the secrets name could be osm-ca-bundle-\<random numbers matching the owning MRC\>?

Reply:
makes sense